packages/server/src/main/java/software/uncharted/terarium/hmiserver/security/SecurityConfig.java

```diff
@@ -11,74 +10,70 @@
 )
 public class SecurityConfig {
-	private final KeycloakLogoutHandler keycloakLogoutHandler;
-
-	@Autowired
-	private KeycloakJwtAuthenticationConverter authenticationConverter;
 
-	@Autowired
-	private Config config;
-
-	SecurityConfig(KeycloakLogoutHandler keycloakLogoutHandler) {
-		this.keycloakLogoutHandler = keycloakLogoutHandler;
-	}
+	private final KeycloakLogoutHandler keycloakLogoutHandler;
+	private final KeycloakJwtAuthenticationConverter authenticationConverter;
+	private final TrustedEndpointRequestMatcher trustedEndpointRequestMatcher;
+	private final SwaggerRequestMatcher swaggerRequestMatcher;
+	private final UnauthenticatedUrlRequestMatcher unauthenticatedUrlRequestMatcher;
 
 	@Bean
 	protected SessionAuthenticationStrategy sessionAuthenticationStrategy() {
 		return new RegisterSessionAuthenticationStrategy(new SessionRegistryImpl());
 	}
 
 	@Bean
-//	@Order(1)
+	// @Order(1)
 	public SecurityFilterChain initialSecurityFilterChain(HttpSecurity http) throws Exception {
-		http
-			.authorizeHttpRequests((authorize) -> {
-			if (config.getEnableSwagger()) {
-				authorize.requestMatchers("/swagger-ui/**", "/swagger-ui.html", "/v3/api-docs/**", "/swagger-resources/**").permitAll();
-			}
-				authorize
-				.requestMatchers(config.getUnauthenticatedUrlPatterns().toArray(new String[0])).permitAll()
-				.requestMatchers(new AntPathRequestMatcher(InternalController.PATH+ "/**")).permitAll()
-				.anyRequest().authenticated();
+		http.authorizeHttpRequests((authorize) -> {
+			authorize
+					.requestMatchers(swaggerRequestMatcher).permitAll()
+					.requestMatchers(unauthenticatedUrlRequestMatcher).permitAll()
+					.requestMatchers(trustedEndpointRequestMatcher).permitAll()
+					.anyRequest().authenticated();
 		});
-		http.oauth2ResourceServer(configurer -> configurer.jwt(jwtConfigurer -> jwtConfigurer.jwtAuthenticationConverter(authenticationConverter)));
+		http.oauth2ResourceServer(configurer -> configurer
+				.jwt(jwtConfigurer -> jwtConfigurer.jwtAuthenticationConverter(authenticationConverter)));
 		return http.build();
 	}
 
-//	@Bean
-//	@Order(2)
-//	public SecurityFilterChain adminSecurityFilterChain(HttpSecurity http) throws Exception {
-//		http
-//			.securityMatcher("/api/users/**")
-//			.authorizeHttpRequests((authorize) -> authorize
-//				.anyRequest().hasRole(Roles.ADMIN)
-//			);
-//		http.oauth2ResourceServer(configurer -> configurer.jwt(jwtConfigurer -> jwtConfigurer.jwtAuthenticationConverter(authenticationConverter)));
-//		return http.build();
-//	}
-//
-//	@Bean
-//	@Order(2)
-//	public SecurityFilterChain userSecurityFilterChain(HttpSecurity http) throws Exception {
-//		http
-//			.securityMatcher("/api/**")
-//			.authorizeHttpRequests((authorize) -> authorize
-//				.anyRequest().hasRole(Roles.USER)
-//			);
-//		http.oauth2ResourceServer(configurer -> configurer.jwt(jwtConfigurer -> jwtConfigurer.jwtAuthenticationConverter(authenticationConverter)));
-//		return http.build();
-//	}
+	// @Bean
```

Comment:
If you want you can totally feel free to delete these Order(2) functions - they are something that should have been deleted a while ago but I was feeling unsure.

Reply:
Done!